src/pages/template.tsx

```diff
@@ -0,0 +1,230 @@
+    setError(inputValue !== "" && !validateInput(inputValue));
+  };
+
+  const handleBlur = () => {
+    setIsFocused(false);
+
+    // Validate input on blur
+    if (value !== "") {
+      setError(!validateInput(value));
+    }
+  };
+
+  const handleFocus = () => {
+    setIsFocused(true);
+  };
+
+  return (
+    <div className="space-y-6">
+      {/* Input Container */}
+      <div className="relative max-w-[320px] font-Montserrat">
+        <Label htmlFor="email" className="text-sm font-medium leading-[20px] text-[#0A3D62]">
+          Email
+        </Label>
+        <div
+          className={`my-2 flex max-h-[64px] items-center rounded-[16px] border px-[24px] py-[20px] text-[16px] font-medium ${
+            error
+              ? isFocused
+                ? "border-[#EF4444] focus-within:ring-4 focus-within:ring-[#FEE2E2]" // Active Error Input
+                : value
+                  ? "border-[#FCA5A5]" // Filled Error Input
+                  : "border-[#FCA5A5]" // Default Error Input
+              : isFocused
+                ? "border-teal-300 focus-within:ring-4 focus-within:ring-[#60E3CF4D]" // Active Input
+                : value
+                  ? "border-[#0A3D624D]" // Filled Input
+                  : "border-[#0A3D624D]" // Default Input
+          }`}
+        >
+          {/* Mail Icon */}
+          <Mail className="text-[#737373]" size={16} />
+
+          {/* Input */}
+          <Input
+            id="email"
+            type="email"
+            placeholder="olivia@untitledui.com"
+            value={value}
+            onChange={handleInputChange}
+            onFocus={handleFocus}
+            onBlur={handleBlur}
+            className="max-h-[24px] flex-1 bg-transparent p-0 pl-2 leading-[24px] focus:outline-none"
+          />
+
+          {/* Question Mark Icon for Hint */}
+          {!error && <HelpCircle size={16} className="text-[#737373]" />}
+
+          {/* Error Icon */}
+          {error && <AlertCircle className="text-[#EF4444]" size={16} />}
+        </div>
+
+        {/* Hint or Error Message */}
+        <p
+          className={`text-xs font-normal leading-[20px] ${error ? "text-[#EF4444]" : "text-[#525252]"}`}
+        >
+          {error ? "This is an error message." : "This is a hint text to help user."}
+        </p>
+      </div>
+    </div>
+  );
+}
+
+export const TypographyStyles = () => {
+  return (
+    <div className="space-y-8 bg-white p-6 text-black">
+      <h1 className="text-2xl font-bold">Typography Styles</h1>
+      <div className="flex justify-around">
+        {/* Montserrat */}
+        <section className="space-y-6">
+          <h2 className="text-xl font-bold">Montserrat</h2>
+          <div className="space-y-4">
+            {/* Regular */}
+            <div>
+              <h3 className="text-lg font-medium">Regular</h3>
+              <div className="space-y-2">
+                <p className="font-Montserrat text-xs font-normal">Montserrat Regular - 12px</p>
```

Comment:
@rakeem996 Likewise here, this needs to ba Typography component which accepts a variant prop.

Header should be h tags while anything that is not a header can be p tags or spans. The designer has fallen short here and he used quite a few variants so let's narrow it down. Take a look at which ones are headers and which ones are used in the body across and design and limit it. Here is how MUI does the variants https://mui.com/material-ui/react-typography/